### PR TITLE
Disable garbage collection during compile_time_instructions count in benchmark base by default.

### PR DESCRIPTION
**Pull request by laithsakka**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #135768

When we measure compile time instruction count, probably we do want in most cases to measure gc instructions
disabling it here by default. 
if it is needed we can add an option to allow it, or someone can use the regular total instruction count instead of compile time instruction count.
cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

File: benchmarks/dynamo/pr_time_benchmarks/benchmark_base.py [hunk 1]
```diff
@@ -117,6 +120,7 @@ def _count_compile_time_instructions(self):
             results.append(count)
 
         config.record_compile_time_instruction_count = False
+        gc.enable()
```
Comment (ezyang) [inline benchmarks/dynamo/pr_time_benchmarks/benchmark_base.py:123]: You need to try...finally this in case of exception?

**Comment by ezyang** [comment]
It's definitely right, from a stability perspective, to disable GC. I do want to punish code that excessively uses lots of memory, since this can increase GC time. But maybe this is not enough of an impact to measure.

**Comment by laithsakka** [comment]
> It's definitely right, from a stability perspective, to disable GC. I do want to punish code that excessively uses lots of memory, since this can increase GC time. But maybe this is not enough of an impact to measure.

Maybe for that a metric that tracks total allocations or peak memory https://docs.python.org/3/library/tracemalloc.html

**Review by ezyang — APPROVED** [review]

File: benchmarks/dynamo/pr_time_benchmarks/benchmarks/basic_modules_benchmarks.py [hunk 2]
```diff
@@ -131,7 +131,8 @@ def name(self):
 
     def _prepare_once(self):
         self.m = self.ModuleClass()
-        self.input = torch.ones(10)
+        torch.set_float32_matmul_precision("high")
```
Comment (laithsakka) [inline benchmarks/dynamo/pr_time_benchmarks/benchmarks/basic_modules_benchmarks.py:134]: ah rebase trash

**Review by anijain2305 — APPROVED** [review]
